pkg/log/checkpoint/fetch.go

```diff
@@ -0,0 +1,72 @@
+package checkpoint
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+type httpClient interface {
+	Get(url string) (resp *http.Response, err error)
+}
+
+func fetchFromUrls(client httpClient, urls []string) []string {
+
+	results := []string{}
+
+	for _, url := range urls {
+		var sb strings.Builder
+		sb.WriteString(fmt.Sprintf("%s: ", url))
+
+		response, err := fetchFromUrl(client, url)
+		if err != nil {
+			sb.WriteString(err.Error())
+		} else {
+			sb.WriteString(response)
+		}
+
+		results = append(results, sb.String())
+	}
+
+	return results
+}
+
+type NotaryRelease struct {
+	Signed struct {
+		Version int `json:"version"`
+	} `json:"signed"`
+}
+
+func fetchFromUrl(client httpClient, url string) (string, error) {
+	response, err := client.Get(url)
```

Comment:
We should have a timeout here

Reply:
the clients are configured with a time out here, https://github.com/James-Pickett/launcher/blob/checkpoint-log-connectivity/pkg/log/checkpoint/checkpoint.go#L53 it applies to all requests made by the client

Reply:
however, I did add a context with time out to the IP lookup, unlike the standard lib http client and dialer, the resolver used for IP look up is not configurable with a time out

Reply:
You're right! There is a timeout in the httpClient. I don't know why one might prefer one over the other, and I think this is fine!